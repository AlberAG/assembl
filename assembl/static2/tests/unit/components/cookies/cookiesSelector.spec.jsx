import React from 'react';
import renderer from 'react-test-renderer';
import CookiesSelector from '../../../../js/app/components/cookies/cookiesSelector';

describe('CookiesSelector component', () => {
  const handleCategorySelectionSpy = jest.fn(() => {});
  const handleSaveSpy = jest.fn(() => {});
  const handleToggleSpy = jest.fn(() => {});

  it('should render a list of cookieToggle components set in different categories', () => {
    const props = {
      cookies: {
        essential: [
          {
            category: 'essential',
            name: 'locale',
            hasChanged: false,
            accepted: true,
            cookieType: 'ACCEPT_LOCALE_ON_SESSION'
          }
        ],
        analytics: [
          {
            category: 'analytics',
            name: 'matomo',
            hasChanged: false,
            accepted: false,
            cookieType: 'REJECT_TRACKING_ON_DISCUSSION'
          }
        ]
      },
      activeKey: 'essential',
      show: true,
      handleCategorySelection: handleCategorySelectionSpy,
      handleSave: handleSaveSpy,
      handleToggle: handleToggleSpy
    };
    const rendered = renderer.create(<CookiesSelector {...props} />);
    expect(rendered).toMatchSnapshot();
  });
});