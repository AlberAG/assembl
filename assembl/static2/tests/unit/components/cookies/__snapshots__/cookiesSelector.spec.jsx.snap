// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CookiesSelector component should render a list of cookieToggle components set in different categories 1`] = `
<div
  className="cookies-selector page-body"
>
  <div>
    <div
      className="cookies-category-selector"
      onClick={[Function]}
    >
      <span
        className="assembl-icon-right-dir active-arrow"
      />
      <span
        className="dark-title-4"
        style={undefined}
      >
        Essential
      </span>
    </div>
    <div
      className="cookies-toggles"
    >
      <div
        className="cookie-toggle"
      >
        <span
          className="cookie-title dark-title-3 ellipsis"
        >
          User Session
        </span>
        <div
          className="rsbc-switch-button rsbc-mode-switch rsbc-switch-button-flat-round disabled"
        >
          <label
            htmlFor="userSession"
          >
            Refuse
          </label>
          <input
            checked={false}
            disabled={true}
            id="userSession"
            name="userSession"
            onChange={[Function]}
            type="checkbox"
            value="1"
          />
          <label
            htmlFor="userSession"
          />
          <label
            htmlFor="userSession"
          >
            Accept
          </label>
        </div>
      </div>
      <div
        className="cookie-toggle"
      >
        <span
          className="cookie-title dark-title-3 ellipsis"
        >
          Language
        </span>
        <div
          className="rsbc-switch-button rsbc-mode-switch rsbc-switch-button-flat-round disabled"
        >
          <label
            htmlFor="locale"
          >
            Refuse
          </label>
          <input
            checked={false}
            disabled={true}
            id="locale"
            name="locale"
            onChange={[Function]}
            type="checkbox"
            value="1"
          />
          <label
            htmlFor="locale"
          />
          <label
            htmlFor="locale"
          >
            Accept
          </label>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="cookies-category-selector"
      onClick={[Function]}
    >
      <span
        className="assembl-icon-right-dir"
      />
      <span
        className="dark-title-4"
        style={undefined}
      >
        Analytique et personnalisation
      </span>
    </div>
    <div
      className="cookies-toggles"
    />
  </div>
  <button
    className="button-submit button-dark btn btn-default"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <span
      className={undefined}
      style={undefined}
    >
      Save
    </span>
  </button>
</div>
`;
